Bug: 部分情况下，加载失败的插件依然会出现在当前已导入的所有插件中

**描述问题：**

通过 nonebot.get_loaded_plugins() 获取当前已导入的所有插件，加载失败的插件也出现在其中

**如何复现？**

1. 复制 2.0.0 a16 版本文档的[注册事件响应器](https://61d3d9dbcadf413fd3238e89--nonebot2.netlify.app/guide/creating-a-matcher.html)一节的 weather.py 的代码作为插件，并在 2.0.0 beta1 版本导入它（很显然，这会发生错误）
2. 通过 nonebot.get_loaded_plugins() 获取已导入的插件，可以发现这个加载失败的插件出现在了已导入的插件中

**期望的结果**

加载失败的插件不出现在已导入的所有插件中

**环境信息：**

 - OS: [Windows]
 - Python Version: [3.10]
 - Nonebot Version: [2.0.0 b1]

**协议端信息：**

 - 协议端: [go-cqhttp]
 - 协议端版本: [1.0.0]

**截图或日志**

```
[ERROR] nonebot | Failed to import "kirami.plugins.weather"
ValueError: Unknown parameter state for function <function handle_first_receive at 0x000001924056AB90> with type typing.Dict[typing.Any, typing.Any]

{Plugin(name='weather', module=<module 'kirami.plugins.weather' from 'D:\\Users\\Documents\\gitee\\KiramiBot\\kirami\\plugins\\weather\\__init__.py'>, module_name='kirami.plugins.weather', manager=<nonebot.plugin.manager.PluginManager object at 0x000002140ECEF130>, export={}, matcher=set(), parent_plugin=None, sub_plugins=set())}
```

--- a/nonebot/plugin/manager.py
+++ b/nonebot/plugin/manager.py
@@ -10,8 +10,8 @@ from typing import Set, Dict, List, Union, Iterable, Optional, Sequence
 
 from nonebot.log import logger
 from nonebot.utils import escape_tag
-from .plugin import Plugin, _new_plugin
 from . import _managers, _current_plugin
+from .plugin import Plugin, _new_plugin, _confirm_plugin
 
 
 class PluginManager:
@@ -179,6 +179,8 @@ class PluginLoader(SourceFileLoader):
         #         f"Error when executing module {module_name} from {module.__file__}."
         #     ) from e
         super().exec_module(module)
+
+        _confirm_plugin(plugin)
 
         _current_plugin.reset(_plugin_token)
         return

--- a/nonebot/plugin/plugin.py
+++ b/nonebot/plugin/plugin.py
@@ -96,5 +96,10 @@ def _new_plugin(fullname: str, module: ModuleType, manager: "PluginManager") -> 
     if name in plugins:
         raise RuntimeError("Plugin already exists! Check your plugin name.")
     plugin = Plugin(name, module, fullname, manager)
-    plugins[name] = plugin
     return plugin
+
+
+def _confirm_plugin(plugin: Plugin) -> None:
+    if plugin.name in plugins:
+        raise RuntimeError("Plugin already exists! Check your plugin name.")
+    plugins[plugin.name] = plugin

--- a/tests/test_init.py
+++ b/tests/test_init.py
@@ -94,3 +94,9 @@ async def test_load_plugin(load_plugin: Set["Plugin"]):
         assert False
     except RuntimeError:
         assert True
+
+    try:
+        nonebot.load_plugin("some_plugin_no_exist")
+        assert False
+    except Exception:
+        assert nonebot.get_plugin("some_plugin_no_exist") is None

